File: code/jvm/src/main/kotlin/pt/isel/leic/ptgest/http/controllers/trainer/model/request/CreateCustomWorkoutRequest.kt
package pt.isel.leic.ptgest.http.controllers.trainer.model.request

import jakarta.validation.constraints.Size
import pt.isel.leic.ptgest.domain.workout.MuscleGroup

data class CreateCustomWorkoutRequest(
    val name: String?,
    val description: String?,
    @field:Size(max = 3, message = "Muscle group list must have at most 3 elements.")
    val muscleGroup: List<MuscleGroup>,
    val sets: List<Int>
) {
    init {
        name?.trim()
        description?.trim()
    }
}

File: code/js/src/views/user/trainerViews/models/workouts/CreateCustomWorkoutRequest.ts
class CreateCustomWorkoutRequest {
    name: string | null
    description: string | null
    muscleGroup: string[]
    sets: number[]

    constructor(name: string | null, description: string | null, muscleGroup: string[], sets: number[]) {
        this.name = name
        this.description = description
        this.muscleGroup = muscleGroup
        this.sets = sets
    }
}
export default CreateCustomWorkoutRequest
